pkg/prometheus/promcfg.go

```diff
@@ -490,10 +490,15 @@ func (cg *configGenerator) generatePodMonitorConfig(
 		}
 	}
 
-	var (
-		relabelings []yaml.MapSlice
-		labelKeys   []string
-	)
+	// Relabel prometheus job name into a meta label
+	relabelings := []yaml.MapSlice{
```

Comment:
Would it make sense to extract this to avoid duplication of code?

Reply:
if it makes sense yes but I wouldn't block the PR on it.